### ISSUE TITLE
Form update background service

### ISSUE THREAD

**Issue by mberg** [onaio/collect#28]
A background service should run checking to see if any forms were updated every 10 minutes.  This will not work if there isn't connectivity. 

If a new form or updated form is detected a standard android alert should be triggered.  Also, the count of new forms/updated forms should update on the home page.
- Explore making this a setting that can be disable.  ie) Run form update background service?
-

**Comment by rowo** [comment]
This means run in the background when the app is open right, not when it's closed? What's wrong with just having it check when the app is launched?

**Comment by rowo** [comment]
Does this need a toggle on/off? Takes up data right?

**Comment by jasonrogena** [comment]
Reenabled this feature. Now using [MQTT](http://mqtt.org/), that should make for faster propagation of updates.

I added two preference though; one for toggling ON/OFF the service (ON by default), and the other for auto downloading forms (OFF by default). A form is auto-downloaded only if it there aren't non-finalised forms on the device.

![screen_2_resize](https://cloud.githubusercontent.com/assets/2384176/21635414/9cd407b8-d26e-11e6-9a88-87d4bb6480ff.png)

Also added a dialog for warning users when they start filling a form that has a pending update. Useful in cases where the notification for an update came through but the device was unable/is still updating the form.

![screen_1_resize](https://cloud.githubusercontent.com/assets/2384176/21635424/a777d316-d26e-11e6-96ad-49587fc8c5ca.png)

**Comment by rowo** [comment]
@jasonrogena What is happening when "Auto check for form updates" is checked? I don't think the label or description is clear. If it is unchecked, how do you check manually?

**Comment by jasonrogena** [comment]
@rowo When the "Auto check for form updates" is checked, the app is constantly checking for form form updates in onaio/onadata. Is '_Constantly check when form updates are made available_' a better description in the settings page?

**Comment by mberg** [comment]
Is this currently working? If so that's awesome

**Comment by jasonrogena** [comment]
It is, but some functionality will have to be built in core https://github.com/onaio/onadata/issues/866 before it can go out

**Comment by rowo** [comment]
@jasonrogena So for "constantly check", does that mean checking when the app is open or running in the background? 

If the app isn't running in the background, it won't check, right?

Is there a frequency that it checks?

**Comment by jasonrogena** [comment]
@rowo It will continue checking for updates in the background, when the app is closed. Hard to attach a 'frequency' to it. It uses the same technology chat apps use to deliver messages, so when an update is made available, all devices with the form should receive the update immediately

**Comment by rowo** [comment]
@jasonrogena by "receive the update immediately" does that mean (1) the form auto-syncs (i.e. gets the new version and replaces the old one) as well or (2) just downloads the form so you can choose whether to update or (3) receives notice that there is an update so the form still has to be grabbed

Asking because I think of this how getting forms works more like email because it's something you fetch/get. Messages you don't really think of as getting them, they just come. So I think most people would think about it like me. So it might be something closer to background check, or form fetch, or sync, depending on what is actually happening here.

**Comment by rowo** [comment]
@jasonrogena Ok I reread your messages I think I get it. It's just a broadcast message that a form update is available. If that's the case, yeah your suggestion might be good. What do you think of:

`Auto check for form updates`
`Constantly check for form updates and be notified when available. Turning off reduces data and power consumption.`

**Comment by jasonrogena** [comment]
@rowo Yeah, it's a notice that there's a new update. The form is only automatically downloaded if the "Auto update forms" preference is checked. Your label and description are much better, thanks

**Comment by bmalina** [comment]
Client has requested for this feature: https://app.intercom.com/a/apps/xzsimrke/respond/inbox/bmalina@ona.io/conversations/9666317330
